GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs

```diff
@@ -68,6 +86,67 @@ private IEnumerable<TestRailSection> GetSectionsTree(ulong projectId, ulong? sui
             return result;
         }
 
+        private void MoveSectionToArchiveRecursively(List<FeatureFileFolder> featureFileFolders,
+            List<TestRailSection> testRailSections)
+        {
+            var archiveSection = _testRailSections.Find(section => section.Name.Equals(_config.TestRailSettings.ArchiveSectionName, StringComparison.InvariantCultureIgnoreCase))?.Id;
+                
+            if (archiveSection is null)
+            {
+                archiveSection = _testRailClientWrapper.CreateSection(new CreateSectionRequest(_config.TestRailSettings.ProjectId,null, _config.TestRailSettings.SuiteId, _config.TestRailSettings.ArchiveSectionName));
+            }
+            
+            foreach (var testRailSection in testRailSections)
+            {
+                var featureFileFolder = featureFileFolders.Find(fileFolder => fileFolder.Name.Equals(testRailSection.Name, StringComparison.InvariantCultureIgnoreCase));
+
+                if (featureFileFolder is not null)
+                {
+                    MoveSectionToArchiveRecursively(featureFileFolder.ChildFolders, testRailSection.ChildSections);
+                    continue;
+                }
+
+                Log.Debug($"Section [{testRailSection.Id.Value}] {testRailSection.Name} is moving to {_config.TestRailSettings.ArchiveSectionName}.");
+                _testRailClientWrapper.MoveSection(testRailSection.Id.Value, archiveSection);
+            }
+        }
+
+        private List<FeatureFileFolder> GetFeatureFileFolderTree(List<IFeatureFile> featureFiles)
+        {
+            var result = new List<FeatureFileFolder>();
+            
+            var featureFilesPaths = featureFiles.Select(file => Path.GetDirectoryName(file.RelativePath)).Distinct().ToArray();
+            
+            var featureFileFoldersDictionary = new Dictionary<string, FeatureFileFolder>();
+
+            foreach (var folderPath in featureFilesPaths)
+            {
+                var pathSeparated = folderPath.Split(Path.DirectorySeparatorChar);
+                var folderName = pathSeparated.LastOrDefault();
```

Comment:
Maybe use just Last() extension method, since you've omitted null-checks for folderName value?

Reply:
Done